src/main/java/com/redhat/devtools/lsp4ij/LSPFileSupport.java

```diff
@@ -118,6 +121,14 @@ public void dispose() {
         getReferenceSupport().cancel();
         getDeclarationSupport().cancel();
         getTypeDefinitionSupport().cancel();
+        var map = getUserMap();
+        var keys= map.getKeys();
+        for(var key : keys) {
+            var value = map.get(key);
+            if (value instanceof Disposable) {
+                ((Disposable) value).dispose();
+            }
+        }

```

Comment:
for (var value : map.values()) {
    if (value instanceof Disposable disposable) {
        disposable.dispose();
    }
}